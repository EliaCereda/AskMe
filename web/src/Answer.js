import React, {Component} from "react";
import Vote from "./Vote";

class Answer extends Component {
    render() {
        return (
            <article className="Answer">

                <header>
                    <h4>answered by {this.props.user}</h4>
                </header>

                <p>{this.props.body}</p>

                <Vote currentUser={this.props.currentUser} answerId={this.props.id} votes={this.props.votes} score={this.props.score} onVote={this.props.onVote} />

            </article>
        );
    }
}

export default Answer;
