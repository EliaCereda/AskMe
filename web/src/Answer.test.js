import React from 'react';
import Answer from "./Answer";

describe('Answer', () => {
    it('renders correctly', () => {
        const answer = {
            body: 'corpo della risposta',
            user: 'utente'
        };

        const wrapper = shallow(<Answer {...answer} />);
        expect(wrapper).toMatchSnapshot();
    });
});
