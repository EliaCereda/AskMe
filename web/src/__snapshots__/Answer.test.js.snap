// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer renders correctly 1`] = `
<article
  className="Answer"
>
  <header>
    <h4>
      answered by 
      utente
    </h4>
  </header>
  <p>
    corpo della risposta
  </p>
  <Vote />
</article>
`;
